code/game/objects/items/weapons/storage/fancy.dm

```diff
@@ -314,6 +314,70 @@
 		new /obj/item/clothing/mask/cigarette(src)
 	create_reagents(15 * storage_slots)//so people can inject cigarettes without opening a packet, now with being able to inject the whole one
 
+/obj/item/weapon/storage/fancy/cigarettes/goldencarp/New()
+	..()
```

Comment:
this is a terrible way of doing it, instead change the behavior in the parent so that it has 
`var/path = /obj/item/clothing/mask/cigarette/goldencarp`
and then it just makes `new path(src)`

then in each subtype, change path (e.g.: for golden carp, `path = /obj/item/clothing/mask/cigarette/goldencarp`

Reply:
done